crates/store/re_datafusion/src/table_entry_provider.rs

```diff
@@ -0,0 +1,84 @@
+use std::sync::Arc;
+
+use async_trait::async_trait;
+
+use arrow::{array::RecordBatch, datatypes::SchemaRef};
+use datafusion::{
+    catalog::TableProvider,
+    error::{DataFusionError, Result as DataFusionResult},
+};
+use re_log_encoding::codec::wire::decoder::Decode as _;
+use re_protos::{
+    common::v1alpha1::ext::EntryId,
+    frontend::v1alpha1::{
+        frontend_service_client::FrontendServiceClient, GetTableSchemaRequest, ScanTableRequest,
+        ScanTableResponse,
+    },
+};
+use tonic::transport::Channel;
+
+use crate::grpc_streaming_provider::{GrpcStreamProvider, GrpcStreamToTable};
+
+#[derive(Debug, Clone)]
+pub struct TableEntryProvider {
```

Comment:
inclined to go verbose and call this `TableEntryTableProvider`. My first reaction was "oh you mean EntryTableProvider" (which is not what this thing actually is).